### ISSUE TITLE
systemd-resolved stoped working with link agregation (bonding)

### ISSUE THREAD

**Issue by a36233** [systemd/systemd#12338]
**systemd version the issue has been seen with**
242

**Used distribution**
Archlinux

**Expected behaviour you didn't see**
 Resolving names

**Unexpected behaviour you saw**
Not resolving names

**Steps to reproduce the problem**
Configure interfaces as bond - active backup and use systemd-networkd and systemd-resolved

$ cat /etc/systemd/network/bond1.netdev 
[NetDev]
Name=bond1
Kind=bond
MACAddress=00:0d:b9:4f:fe:44

[Bond]
Mode=active-backup
LACPTransmitRate=fast
MIIMonitorSec=1s
UpDelaySec=100ms
DownDelaySec=100ms

$ cat /etc/systemd/network/enpXs0.network
[Match]
Name=enp*

[Network]
Bond=bond1

$ cat /etc/systemd/network/bond1.network 
[Match]
Name=bond1

[Network]
DHCP=yes

[DHCP]
RouteMetric=10

$ networkctl 
IDX LINK             TYPE               OPERATIONAL      SETUP     
  1 lo               loopback           carrier          unmanaged 
  2 enp1s0           ether              enslaved         configured
  3 enp2s0           ether              enslaved         configured
  5 enp3s0           ether              no-carrier       configuring
  6 wlp4s0           wlan               no-carrier       configuring
  7 bond1            bond               degraded-carrier configured
  8 wwp0s19u1u3c3    wwan               off              unmanaged 

$ resolvectl
Global
       LLMNR setting: yes
MulticastDNS setting: yes
  DNSOverTLS setting: no
      DNSSEC setting: allow-downgrade
    DNSSEC supported: yes
Fallback DNS Servers: 1.1.1.1
                      9.9.9.10
                      8.8.8.8
                      2606:4700:4700::1111
                      2620:fe::10
                      2001:4860:4860::8888
          DNSSEC NTA: 10.in-addr.arpa
                      16.172.in-addr.arpa
                      168.192.in-addr.arpa
                      17.172.in-addr.arpa
                      18.172.in-addr.arpa
                      19.172.in-addr.arpa
                      20.172.in-addr.arpa
                      21.172.in-addr.arpa
                      22.172.in-addr.arpa
                      23.172.in-addr.arpa
                      24.172.in-addr.arpa
                      25.172.in-addr.arpa
                      26.172.in-addr.arpa
                      27.172.in-addr.arpa
                      28.172.in-addr.arpa
                      29.172.in-addr.arpa
                      30.172.in-addr.arpa
                      31.172.in-addr.arpa
                      corp
                      d.f.ip6.arpa
                      home
                      internal
                      intranet
                      lan
                      local
                      private
                      test

Link 8 (wwp0s19u1u3c3)
      Current Scopes: none
DefaultRoute setting: no
       LLMNR setting: yes
MulticastDNS setting: no
  DNSOverTLS setting: no
      DNSSEC setting: allow-downgrade
    DNSSEC supported: yes

Link 7 (bond1)
      Current Scopes: none
DefaultRoute setting: no
       LLMNR setting: yes
MulticastDNS setting: no
  DNSOverTLS setting: no
      DNSSEC setting: allow-downgrade
    DNSSEC supported: yes
  Current DNS Server: 193.136.92.73
         DNS Servers: 193.136.92.73
                      193.136.92.74

Link 6 (wlp4s0)
      Current Scopes: none
DefaultRoute setting: no
       LLMNR setting: yes
MulticastDNS setting: no
  DNSOverTLS setting: no
      DNSSEC setting: allow-downgrade
    DNSSEC supported: yes

Link 5 (enp3s0)
      Current Scopes: none
DefaultRoute setting: no
       LLMNR setting: yes
MulticastDNS setting: no
  DNSOverTLS setting: no
      DNSSEC setting: allow-downgrade
    DNSSEC supported: yes

Link 3 (enp2s0)
      Current Scopes: none
DefaultRoute setting: no
       LLMNR setting: yes
MulticastDNS setting: no
  DNSOverTLS setting: no
      DNSSEC setting: allow-downgrade
    DNSSEC supported: yes

Link 2 (enp1s0)
      Current Scopes: none
DefaultRoute setting: no
       LLMNR setting: yes
MulticastDNS setting: no
  DNSOverTLS setting: no
      DNSSEC setting: allow-downgrade
    DNSSEC supported: yes

**Comment by yuwata** [comment]
I guess this is a duplicate of #12315. Could you test PR #12316 or the package linked at https://github.com/systemd/systemd/issues/12315#issuecomment-483887718?

**Comment by Gigadoc2** [comment]
I tried the patched package, it did not seem to change anything for me.

Output of **systemd-networkd** after restarting with debug messages:
```
Apr 25 10:07:20 vash systemd[1]: Stopping Network Service...
Apr 25 10:07:20 vash systemd[1]: systemd-networkd.service: Succeeded.
Apr 25 10:07:20 vash systemd[1]: Stopped Network Service.
Apr 25 10:07:20 vash systemd[1]: Starting Network Service...
Apr 25 10:07:21 vash systemd-networkd[3477]: Bus bus-api-network: changing state UNSET → OPENING
Apr 25 10:07:21 vash systemd-networkd[3477]: Bus bus-api-network: changing state OPENING → AUTHENTICATING
Apr 25 10:07:21 vash systemd-networkd[3477]: timestamp of '/etc/systemd/network' changed
Apr 25 10:07:21 vash systemd-networkd[3477]: timestamp of '/usr/lib/systemd/network' changed
Apr 25 10:07:21 vash systemd-networkd[3477]: No virtualization found in DMI
Apr 25 10:07:21 vash systemd-networkd[3477]: No virtualization found in CPUID
Apr 25 10:07:21 vash systemd-networkd[3477]: Virtualization XEN not found, /proc/xen does not exist
Apr 25 10:07:21 vash systemd-networkd[3477]: This platform does not support /proc/device-tree
Apr 25 10:07:21 vash systemd-networkd[3477]: UML virtualization not found in /proc/cpuinfo.
Apr 25 10:07:21 vash systemd-networkd[3477]: This platform does not support /proc/sysinfo
Apr 25 10:07:21 vash systemd-networkd[3477]: Found VM virtualization none
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: loaded bond
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Creating
Apr 25 10:07:21 vash systemd-networkd[3477]: /usr/lib/systemd/network/80-container-host0.network: Conditions in the file do not match the system environment, skipping.
Apr 25 10:07:21 vash systemd-networkd[3477]: wlp3s0: Flags change: +SLAVE +UP +LOWER_UP +RUNNING +MULTICAST +BROADCAST
Apr 25 10:07:21 vash systemd-networkd[3477]: wlp3s0: Link 4 added
Apr 25 10:07:21 vash systemd-networkd[3477]: wlp3s0: udev initialized link
Apr 25 10:07:21 vash systemd-networkd[3477]: wlp3s0: State changed: pending -> initialized
Apr 25 10:07:21 vash systemd-networkd[3477]: wlp3s0: Saved original MTU: 1500
Apr 25 10:07:21 vash systemd-networkd[3477]: enp0s25: Flags change: +SLAVE +UP +MULTICAST +BROADCAST
Apr 25 10:07:21 vash systemd-networkd[3477]: enp0s25: Link 3 added
Apr 25 10:07:21 vash systemd-networkd[3477]: enp0s25: udev initialized link
Apr 25 10:07:21 vash systemd-networkd[3477]: enp0s25: State changed: pending -> initialized
Apr 25 10:07:21 vash systemd-networkd[3477]: enp0s25: Saved original MTU: 1500
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: New device has no master, continuing without
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Flags change: +MASTER +UP +LOWER_UP +RUNNING +MULTICAST +BROADCAST
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Link 2 added
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: udev initialized link
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: State changed: pending -> initialized
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: netdev has index 2
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: netdev ready
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Saved original MTU: 1500
Apr 25 10:07:21 vash systemd-networkd[3477]: lo: New device has no master, continuing without
Apr 25 10:07:21 vash systemd-networkd[3477]: lo: Flags change: +LOOPBACK +UP +LOWER_UP +RUNNING
Apr 25 10:07:21 vash systemd-networkd[3477]: lo: Link 1 added
Apr 25 10:07:21 vash systemd-networkd[3477]: lo: udev initialized link
Apr 25 10:07:21 vash systemd-networkd[3477]: lo: State changed: pending -> initialized
Apr 25 10:07:21 vash systemd-networkd[3477]: lo: Saved original MTU: 65536
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Adding address: fe80::[redacted]/64 (valid forever)
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Gained IPv6LL
Apr 25 10:07:21 vash systemd-networkd[3477]: lo: Adding address: ::1/128 (valid forever)
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Adding address: 145.[redacted]/22 (valid for 47min 25s)
Apr 25 10:07:21 vash systemd-networkd[3477]: lo: Adding address: 127.0.0.1/8 (valid forever)
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Adding route: dst: ff00::/8, src: n/a, gw: n/a, prefsrc: n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Adding route: dst: fe80::[redacted]/128, src: n/a, gw: n/a, prefsrc: n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: lo: Adding route: dst: ::1/128, src: n/a, gw: n/a, prefsrc: n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Adding route: dst: fe80::/64, src: n/a, gw: n/a, prefsrc: n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: lo: Adding route: dst: ::1/128, src: n/a, gw: n/a, prefsrc: n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Adding route: dst: 145.[redacted]/32, src: n/a, gw: n/a, prefsrc: 145.[redacted]
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Adding route: dst: 145.[redacted]/32, src: n/a, gw: n/a, prefsrc: 145.[redacted]
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Adding route: dst: 145.[redacted]/32, src: n/a, gw: n/a, prefsrc: 145.[redacted]
Apr 25 10:07:21 vash systemd-networkd[3477]: lo: Adding route: dst: 127.255.255.255/32, src: n/a, gw: n/a, prefsrc: 127.0.0.1
Apr 25 10:07:21 vash systemd-networkd[3477]: lo: Adding route: dst: 127.0.0.1/32, src: n/a, gw: n/a, prefsrc: 127.0.0.1
Apr 25 10:07:21 vash systemd-networkd[3477]: lo: Adding route: dst: 127.0.0.0/8, src: n/a, gw: n/a, prefsrc: 127.0.0.1
Apr 25 10:07:21 vash systemd-networkd[3477]: lo: Adding route: dst: 127.0.0.0/32, src: n/a, gw: n/a, prefsrc: 127.0.0.1
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Adding route: dst: 145.[redacted]/32, src: n/a, gw: n/a, prefsrc: 145.[redacted]
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Adding route: dst: 145.[redacted]/22, src: n/a, gw: n/a, prefsrc: 145.[redacted]
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Adding route: dst: n/a, src: n/a, gw: 145.[redacted], prefsrc: 145.[redacted]
Apr 25 10:07:21 vash systemd-networkd[3477]: rtnl: received address with invalid family 129, ignoring
Apr 25 10:07:21 vash systemd-networkd[3477]: rtnl: received address with invalid family 128, ignoring
Apr 25 10:07:21 vash systemd-networkd[3477]: Enumeration completed
Apr 25 10:07:21 vash systemd-networkd[3477]: Bus bus-api-network: changing state AUTHENTICATING → HELLO
Apr 25 10:07:21 vash systemd[1]: Started Network Service.
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=Hello cookie=1 reply_cookie=0 signature=n/a error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=RequestName cookie=2 reply_cookie=0 signature=su error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=AddMatch cookie=3 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_34 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=4 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_34 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=5 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_33 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=6 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_33 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=7 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=8 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=9 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_31 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=10 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_31 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=11 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=12 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=13 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=14 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: netdev exists, using existing without changing its parameters
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Created
Apr 25 10:07:21 vash systemd-networkd[3477]: wlp3s0: Link state is up-to-date
Apr 25 10:07:21 vash systemd-networkd[3477]: wlp3s0: found matching network '/etc/systemd/network/14-wlp3s0.network'
Apr 25 10:07:21 vash systemd-networkd[3477]: Setting '/proc/sys/net/ipv6/conf/wlp3s0/proxy_ndp' to '0'
Apr 25 10:07:21 vash systemd-networkd[3477]: Setting '/proc/sys/net/ipv6/conf/wlp3s0/use_tempaddr' to '0'
Apr 25 10:07:21 vash systemd-networkd[3477]: Setting '/proc/sys/net/ipv6/conf/wlp3s0/accept_ra' to '0'
Apr 25 10:07:21 vash systemd-networkd[3477]: LLDP: Started LLDP client
Apr 25 10:07:21 vash systemd-networkd[3477]: wlp3s0: Started LLDP.
Apr 25 10:07:21 vash systemd-networkd[3477]: wlp3s0: Setting address genmode for link
Apr 25 10:07:21 vash systemd-networkd[3477]: wlp3s0: State changed: initialized -> configuring
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_34 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=15 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: wlp3s0: Configured
Apr 25 10:07:21 vash systemd-networkd[3477]: wlp3s0: State changed: configuring -> configured
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_34 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=16 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Got message type=method_return sender=org.freedesktop.DBus destination=:1.128 path=n/a interface=n/a member=n/a cookie=1 reply_cookie=1 signature=s error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Bus bus-api-network: changing state HELLO → RUNNING
Apr 25 10:07:21 vash systemd-networkd[3477]: Got message type=signal sender=org.freedesktop.DBus.Local destination=n/a path=/org/freedesktop/DBus/Local interface=org.freedesktop.DBus.Local member=Connected cookie=4294967295 reply_cookie=0 signature=n/a error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Got message type=signal sender=org.freedesktop.DBus destination=:1.128 path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=NameAcquired cookie=2 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Got message type=signal sender=org.freedesktop.DBus destination=:1.128 path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=NameAcquired cookie=3 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Got message type=method_return sender=org.freedesktop.DBus destination=:1.128 path=n/a interface=n/a member=n/a cookie=4 reply_cookie=2 signature=u error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Successfully acquired requested service name.
Apr 25 10:07:21 vash systemd-networkd[3477]: Got message type=method_return sender=org.freedesktop.DBus destination=:1.128 path=n/a interface=n/a member=n/a cookie=5 reply_cookie=3 signature=n/a error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Match type='signal',sender='org.freedesktop.login1',path='/org/freedesktop/login1',interface='org.freedesktop.login1.Manager',member='PrepareForSleep' successfully installed.
Apr 25 10:07:21 vash systemd-networkd[3477]: enp0s25: Link state is up-to-date
Apr 25 10:07:21 vash systemd-networkd[3477]: enp0s25: found matching network '/etc/systemd/network/15-enp0s25.network'
Apr 25 10:07:21 vash systemd-networkd[3477]: Setting '/proc/sys/net/ipv6/conf/enp0s25/proxy_ndp' to '0'
Apr 25 10:07:21 vash systemd-networkd[3477]: Setting '/proc/sys/net/ipv6/conf/enp0s25/use_tempaddr' to '0'
Apr 25 10:07:21 vash systemd-networkd[3477]: Setting '/proc/sys/net/ipv6/conf/enp0s25/accept_ra' to '0'
Apr 25 10:07:21 vash systemd-networkd[3477]: LLDP: Started LLDP client
Apr 25 10:07:21 vash systemd-networkd[3477]: enp0s25: Started LLDP.
Apr 25 10:07:21 vash systemd-networkd[3477]: enp0s25: Setting address genmode for link
Apr 25 10:07:21 vash systemd-networkd[3477]: enp0s25: State changed: initialized -> configuring
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_33 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=17 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Link state is up-to-date
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: found matching network '/etc/systemd/network/20-bond0.network'
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Removing address 145.[redacted]
Apr 25 10:07:21 vash systemd-networkd[3477]: Setting '/proc/sys/net/ipv6/conf/bond0/proxy_ndp' to '0'
Apr 25 10:07:21 vash systemd-networkd[3477]: Setting '/proc/sys/net/ipv6/conf/bond0/use_tempaddr' to '2'
Apr 25 10:07:21 vash systemd-networkd[3477]: Setting '/proc/sys/net/ipv6/conf/bond0/accept_ra' to '0'
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Setting address genmode for link
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Acquiring DHCPv4 lease
Apr 25 10:07:21 vash systemd-networkd[3477]: DHCP CLIENT (0x[redacted]): STARTED on ifindex 2
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Discovering IPv6 routers
Apr 25 10:07:21 vash systemd-networkd[3477]: NDISC: Started IPv6 Router Solicitation client
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: State changed: initialized -> configuring
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=18 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: NDISC: Sent Router Solicitation, next solicitation in 3s
Apr 25 10:07:21 vash systemd-networkd[3477]: DHCP CLIENT (0x[redacted]): DISCOVER
Apr 25 10:07:21 vash systemd-networkd[3477]: lo: Link state is up-to-date
Apr 25 10:07:21 vash systemd-networkd[3477]: lo: Unmanaged
Apr 25 10:07:21 vash systemd-networkd[3477]: lo: State changed: initialized -> unmanaged
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_31 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=19 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Removing address: 145.[redacted]/22 (valid for 47min 24s)
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=20 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=21 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Removing route: dst: 145.[redacted]/22, src: n/a, gw: n/a, prefsrc: 145.[redacted]
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Removing route: dst: 145.[redacted]/32, src: n/a, gw: n/a, prefsrc: 145.[redacted]
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Removing route: dst: 145.[redacted]/32, src: n/a, gw: n/a, prefsrc: 145.[redacted]
Apr 25 10:07:21 vash systemd-networkd[3477]: bond0: Removing route: dst: 145.[redacted]/32, src: n/a, gw: n/a, prefsrc: 145.[redacted]
Apr 25 10:07:22 vash systemd-networkd[3477]: DHCP CLIENT (0x[redacted]): OFFER
Apr 25 10:07:22 vash systemd-networkd[3477]: DHCP CLIENT (0x[redacted]): REQUEST (requesting)
Apr 25 10:07:22 vash systemd-networkd[3477]: DHCP CLIENT (0x[redacted]): ACK
Apr 25 10:07:22 vash systemd-networkd[3477]: DHCP CLIENT (0x[redacted]): lease expires in 59min 57s
Apr 25 10:07:22 vash systemd-networkd[3477]: DHCP CLIENT (0x[redacted]): T2 expires in 52min 27s
Apr 25 10:07:22 vash systemd-networkd[3477]: DHCP CLIENT (0x[redacted]): T1 expires in 29min 58s
Apr 25 10:07:22 vash systemd-networkd[3477]: bond0: DHCPv4 address 145.[redacted]/22 via 145.[redacted]
Apr 25 10:07:22 vash systemd-networkd[3477]: Setting transient hostname: 'vash'
Apr 25 10:07:22 vash systemd-networkd[3477]: Sent message type=method_call sender=n/a destination=org.freedesktop.hostname1 path=/org/freedesktop/hostname1 interface=org.freedesktop.hostname1 member=SetHostname cookie=22 reply_cookie=0 signature=sb error-name=n/a error-message=n/a
Apr 25 10:07:22 vash systemd-networkd[3477]: bond0: Updating address: 145.[redacted]/22 (valid for 1h)
Apr 25 10:07:22 vash systemd-networkd[3477]: bond0: Adding route: dst: 145.[redacted]/32, src: n/a, gw: n/a, prefsrc: 145.[redacted]
Apr 25 10:07:22 vash systemd-networkd[3477]: bond0: Adding route: dst: 145.[redacted]/32, src: n/a, gw: n/a, prefsrc: 145.[redacted]
Apr 25 10:07:22 vash systemd-networkd[3477]: bond0: Adding route: dst: 145.[redacted]/22, src: n/a, gw: n/a, prefsrc: 145.[redacted]
Apr 25 10:07:22 vash systemd-networkd[3477]: bond0: Adding route: dst: 145.[redacted]/32, src: n/a, gw: n/a, prefsrc: 145.[redacted]
Apr 25 10:07:22 vash systemd-networkd[3477]: bond0: DHCP: No routes received from DHCP server: No data available
Apr 25 10:07:22 vash systemd-networkd[3477]: bond0: Configuring route: dst: 145.[redacted]/32, src: n/a, gw: n/a, prefsrc: 145.[redacted]
Apr 25 10:07:22 vash systemd-networkd[3477]: bond0: Configuring route: dst: n/a, src: n/a, gw: 145.[redacted], prefsrc: 145.[redacted]
Apr 25 10:07:22 vash systemd-networkd[3477]: bond0: Updating route: dst: 145.[redacted]/32, src: n/a, gw: n/a, prefsrc: 145.[redacted]
Apr 25 10:07:22 vash systemd-networkd[3477]: bond0: Updating route: dst: n/a, src: n/a, gw: 145.[redacted], prefsrc: 145.[redacted]
Apr 25 10:07:22 vash systemd-networkd[3477]: Got message type=method_return sender=:1.129 destination=:1.128 path=n/a interface=n/a member=n/a cookie=3 reply_cookie=22 signature=n/a error-name=n/a error-message=n/a
Apr 25 10:07:25 vash systemd-networkd[3477]: NDISC: Sent Router Solicitation, next solicitation in 8s
```

**resolvectl**
```
Global
       LLMNR setting: yes
MulticastDNS setting: yes
  DNSOverTLS setting: no
      DNSSEC setting: allow-downgrade
    DNSSEC supported: no
Fallback DNS Servers: 1.1.1.1
                      9.9.9.10
                      8.8.8.8
                      2606:4700:4700::1111
                      2620:fe::10
                      2001:4860:4860::8888
          DNSSEC NTA: 10.in-addr.arpa
                      16.172.in-addr.arpa
                      168.192.in-addr.arpa
                      17.172.in-addr.arpa
                      18.172.in-addr.arpa
                      19.172.in-addr.arpa
                      20.172.in-addr.arpa
                      21.172.in-addr.arpa
                      22.172.in-addr.arpa
                      23.172.in-addr.arpa
                      24.172.in-addr.arpa
                      25.172.in-addr.arpa
                      26.172.in-addr.arpa
                      27.172.in-addr.arpa
                      28.172.in-addr.arpa
                      29.172.in-addr.arpa
                      30.172.in-addr.arpa
                      31.172.in-addr.arpa
                      corp
                      d.f.ip6.arpa
                      dn42
                      hack
                      home
                      internal
                      intranet
                      lan
                      local
                      private
                      test

Link 4 (wlp3s0)
      Current Scopes: none
DefaultRoute setting: no
       LLMNR setting: yes
MulticastDNS setting: no
  DNSOverTLS setting: no
      DNSSEC setting: allow-downgrade
    DNSSEC supported: yes

Link 3 (enp0s25)
      Current Scopes: none
DefaultRoute setting: no
       LLMNR setting: yes
MulticastDNS setting: no
  DNSOverTLS setting: no
      DNSSEC setting: allow-downgrade
    DNSSEC supported: yes

Link 2 (bond0)
      Current Scopes: none
DefaultRoute setting: yes
       LLMNR setting: yes
MulticastDNS setting: resolve
  DNSOverTLS setting: no
      DNSSEC setting: no
    DNSSEC supported: no
         DNS Servers: 131.174.117.20
```

**networkctl**
```
IDX LINK             TYPE               OPERATIONAL      SETUP
  1 lo               loopback           carrier          unmanaged
  2 bond0            bond               degraded-carrier configured
  3 enp0s25          ether              no-carrier       configuring
  4 wlp3s0           wlan               enslaved         configured

4 links listed.
```

**10-bond0.netdev**
```
/etc/systemd/network/10-bond0.netdev
[Match]
Host=a250c95b2c44480d8d5f6926185bc349
Virtualization=no

[NetDev]
Description=Bonded Internet Uplink
Name=bond0
Kind=bond

[Bond]
Mode=active-backup
MIIMonitorSec=100ms
UpDelaySec=500ms
FailOverMACPolicy=none
AllSlavesActive=1
```

**/etc/systemd/network/14-wlp3s0.network**
```
[Match]
Host=a250c95b2c44480d8d5f6926185bc349
Virtualization=no
Name=wlp3s0

[Network]
Description=WiFi uplink slave
Bond=bond0
LLDP=true
```

**15-enp0s25.network**
```
[Match]
Host=a250c95b2c44480d8d5f6926185bc349
Virtualization=no
Name=enp0s25

[Network]
Description=Ethernet uplink slave
Bond=bond0
PrimarySlave=true
LLDP=true
EmitLLDP=true
```

**20-bond0.network**
```
[Match]
Host=a250c95b2c44480d8d5f6926185bc349
Virtualization=no
Type=bond
Name=bond0

[Network]
Description=Bonded internet uplink
DHCP=yes
#LinkLocalAddressing=yes
#IPv4LLRoute=yes
LLMNR=yes
MulticastDNS=resolve
#DNSSEC=allow-downgrade
DNSSEC=no
LLDP=no
BindCarrier=wlp3s0 enp0s25
#Domains=~.
DNSDefaultRoute=true
IPv6PrivacyExtensions=yes
IPv6AcceptRA=yes

[DHCP]
UseDomains=true

[IPv6AcceptRA]
UseDomains=true
```

**Comment by poettering** [comment]
I guess this is a duplicate of #12285. Fixed by #12381
